samples/network/httptest/source/http.c

```diff
@@ -0,0 +1,319 @@
+#include <http/https.h>
+#include <ssl/ssl.h>
+#include <net/net.h>
+#include <sysmodule/sysmodule.h>
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#define HTTP_YES		1
+#define HTTP_NO 		0
+#define HTTP_SUCCESS 	1
+#define HTTP_FAILED	 	0
+
+#define HTTP_USER_AGENT "Mozilla/5.0 (PLAYSTATION 3; 1.00)"
+
+typedef struct
+{
+    void* http_pool;
+    void* ssl_pool;
+    httpsData* caList;
+    void* cert_buffer;
+} t_http_pools;
+
+static t_http_pools http_pools;
+u8 cancel = HTTP_NO;
+
+static char getBuffer[64*1024];
+
+
+int http_init(void)
+{
+    int ret;
+	u32 cert_size=0;
+	u8 module_https_loaded=HTTP_NO;
+	u8 module_http_loaded=HTTP_NO;
+	u8 module_net_loaded=HTTP_NO;
+	u8 module_ssl_loaded=HTTP_NO;
+	
+	u8 https_init=HTTP_NO;
+	u8 http_init=HTTP_NO;
+	u8 net_init=HTTP_NO;
+	u8 ssl_init=HTTP_NO;
+
+	//init
+	ret = sysModuleLoad(SYSMODULE_NET);
```

Comment:
`netInitialize()` does the module loading for you, so you shouldn't need to do this + netInitialize() in theory?
```
s32 netInitialize()
{
	//....
	
	ret = sysModuleLoad(SYSMODULE_NET);
	if(ret<0) return lv2errno(ret);
        
        //....

	return ret;
}
```

Reply:
yes, I know it's not needed, but my criteria for samples is to be as verbose and explicit as possible. My goal here is (hopefully 🙏 ) to help someone to learn, not to achieve the best http-fetch library. 😄 
I think that an example should be easy to understand for any newcomer who doesn't have a clue about the internals of a library or SDK. (E.g., If I remove the line, the user might never know there's a `SYSMODULE_NET` at all)

In that sense, the code is surely un-optimized in many ways (it uses many variables, prints messages for each error, etc.), but hopefully it will give an overview of the main stuff available in `libhttp`.

Anyways, all feedback is appreciated 👍

Reply:
lol ok! I wasn't sure if it could potentially be an issue for an app to try and load a sys module multiple times!

Reply:
No, actually that's one of the things SCE took care in their design. You can request to load modules many times, and the system takes care of dependencies and all.
So for example: if you try to load `SYSMODULE_HTTPS` directly (and nothing was loaded before), the PS3 will load NET, HTTP, and SSL modules along with HTTPS. 
For unloading, the system also check if the modules are still in use or have cross dependencies, to avoid issues. (another nice feature)

Reply:
Oh nice that's really good to know, thanks!